src/main/webapp/app/entities/modeling-exercise/modeling-exercise.route.ts

```diff
@@ -26,46 +52,50 @@ export const modelingExerciseRoute: Routes = [
         canActivate: [UserRouteAccessService],
     },
     {
-        path: 'course/:courseId/modeling-exercise',
-        component: ModelingExerciseComponent,
+        path: 'course/:courseId/modeling-exercise/new',
+        component: ModelingExerciseUpdateComponent,
+        resolve: {
+            modelingExercise: ModelingExerciseResolver,
+        },
         data: {
             authorities: ['ROLE_TA', 'ROLE_INSTRUCTOR', 'ROLE_ADMIN'],
             pageTitle: 'artemisApp.modelingExercise.home.title',
         },
         canActivate: [UserRouteAccessService],
     },
     {
-        path: 'course/:courseId/modeling-exercise/:id',
-        component: ModelingExerciseDetailComponent,
+        path: 'modeling-exercise/:exerciseId/edit',
+        component: ModelingExerciseUpdateComponent,
+        resolve: {
+            modelingExercise: ModelingExerciseResolver,
+        },
         data: {
             authorities: ['ROLE_TA', 'ROLE_INSTRUCTOR', 'ROLE_ADMIN'],
             pageTitle: 'artemisApp.modelingExercise.home.title',
         },
         canActivate: [UserRouteAccessService],
     },
-];
-
-export const modelingExercisePopupRoute: Routes = [
     {
-        path: 'course/:courseId/modeling-exercise-new',
-        component: ModelingExercisePopupComponent,
+        path: 'course/:courseId/modeling-exercise',
```

Comment:
I have a quick question: are the paths`course/:courseId/modeling-exercise` and `course/:courseId/modeling-exercise/:id` called somewhere? If not, then you can just delete these paths imo :)

Reply:
Yes, they are still used. However the path `course/:courseId/modeling-exercise`, so maybe I will get rid of it

Reply:
However I removed `modeling-exercise` route since it is not used